app/models/course/assessment/submission.rb

```diff
@@ -10,7 +10,8 @@ class Course::Assessment::Submission < ActiveRecord::Base
 
   workflow do
     state :attempting do
-      event :finalise, transitions_to: :submitted
+      event :finalise, transitions_to: :graded, if: proc { |s| s.automatically_release_grades? }
```

Comment:
Should the transition to `graded` only be made after we have ensured that all the auto-gradings are in?
Instead of jumping straight to `graded`, perhaps it would be better allow the transition to `submitted` (which triggers the auto-grading) then callback to `publish!` once the submission grading is done.

If we go ahead transitioning straight to `graded`, would it be good to add `auto_grade! if automatically_release_grades?` to `#finalise` in this commit?